classes/class.ilMumieTaskSSOToken.php

```diff
@@ -115,6 +115,26 @@ public static function invalidateTokenForUser($userId)
         $mumie_token->delete();
     }
 
+    public static function invalidateAllTokenForUser($userId)
+    {
+        global $ilDB;
+        include_once('Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask/classes/class.ilMumieTaskIdHashingService.php');
+        $hashedId = ilMumieTaskIdHashingService::getHashForUser($userId);
+        $query = "SELECT * FROM "
+        . self::MUMIETOKENS_TABLE_NAME
+        . " WHERE user LIKE "
+        . $ilDB->quote($hashedId . "%", 'text');
+        $result = $ilDB->fetchAssoc($ilDB->query($query));
+        if (!is_null($result["timecreated"]) && !is_null($result["token"])) {
+            while($result = $ilDB->fetchAssoc($ilDB->query($query))) {
+                $mumie_token = new ilMumieTaskSSOToken($result["user"]);
+                $mumie_token->delete();
+            }
+            return true;
+        }
+        return false;
```

Comment:
I don't think that this function should have a return value as it's name doesn't suggest a return value. I also don't think that it's actually necessary for MumieTaskHook to begin with.